### DISCUSSION TITLE
How to run multiple Mace models on one GPU

### DISCUSSION THREAD

**Discussion by Swiftily** [ACEsuit/mace#440] [Q&A] [answered]
Is it possible to run multiple mace models on a single GPU? I am using a mace model as a driver for i-PI to perform MD simulations, but I only have one GPU and I found that running one model occupies few memory of GPU. Therefore, I am wondering if it is possible to run multiple Mace instances on one GPU. Thank you all for your answers, the following is my mace_driver.py file which launches mace.
```from mace.calculators import MACECalculator

from ase.calculators.socketio import SocketClient
from ase.io import read


atoms = read("../init_and_model/a.xyz") 


mace_calc = MACECalculator(
        model_paths=['../init_and_model/MACE-OFF23_small.model'],
        device='cpu',
        default_dtype='float64', 
        )

atoms.set_calculator(mace_calc)


host = 'mace1'
client = SocketClient(unixsocket=host)

client.run(atoms)

**Comment by stargolike** [comment]
I encountered this situation where you want to run two tasks for ASE (Atomic Simulation Environment). I think you can use some commands like 'nohup' to run the processes in the background

**Comment by ilyes319** [comment]
I think @RokasEl knows how to do that.

**Comment by RokasEl** [comment, marked as answer]
For small simulations usually I just launch a bunch of jobs in a for loop in the background. For example:

```bash
for idx in $(seq 0 1 16); do
    python ./run_water_md.py &
done
wait
```

Where `run_water_md.py` is just a small script that uses ASE calculators etc. Depending on your cluster configuration, you might need to enable the CUDA multi-process service using

```bash
nvidia-cuda-mps-control -d
```

at the start of your bash script. See [link](https://docs.nvidia.com/deploy/mps/index.html) for more info.
